knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -380,3 +407,33 @@ func (r *ReconcileKnativeKafka) deleteKnativeKafkaSource(instance *operatorv1alp
 	}
 	return nil
 }
+
+func (r *ReconcileKnativeKafka) buildManifest(instance *operatorv1alpha1.KnativeKafka) (*mf.Manifest, error) {
+	combinedManifest := &mf.Manifest{}
```

Comment:
This made me pause a few times because I was figuring out what gets mutated when. Maybe we should be more explicit like so:

```golang
var resources []*unstructured.Unstructured

if instance.Spec.Channel.Enabled {
  resources = append(resources, r.rawKafkaChannelManifest.Resources())
}

if instance.Spec.Source.Enabled {
  resources = append(resources, r.rawKafkaSourceManifest.Resources())
}

result, err := mf.ManifestFrom(mf.Slice(append(m1.Resources(), m2.Resources()...)))
if err != nil {
	return nil, fmt.Errorf("failed to merge manifests: %w", err)
}

combined, err := mf.ManifestFrom(mf.Slice(resources), mf.UseClient(rclient))
...
```

(also note the `UseClient` helper, we probably should also add a logger to the new manifest)

Reply:
Thx for the feedback. Pushed a new commit, looks much better now.